h/admin/services/user.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+from h import models
+from h.api.search import index
+from h.util.user import userid_from_username
+
+
+class UserRenameError(Exception):
+    pass
+
+
+class RenameUserService(object):
+    """
+    Renames a user and updates all its annotations and when necessary the NIPSA
+    status.
+
+    ``check`` should be called first
+
+    Validates the new username and updates the User. The permissions of
```

Comment:
The permissions? I think it would be better to say `userid`, as that's what's actually being updated.